device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1711,62 +1763,28 @@ private void addToWaitingQueue(IotHubTransportPacket packet)
             log.warn("Exception thrown while calling the onQueueRequest callback in addToWaitingQueue", ex);
         }
 
-        synchronized (this.sendThreadLock)
-        {
-            this.waitingPacketsQueue.add(packet);
+        this.waitingPacketsQueue.add(packet);
 
-            // Wake up IotHubSendTask so it can send this message
-            this.sendThreadLock.notifyAll();
-        }
+        // Wake up IotHubSendTask so it can send this message
+        this.sendThreadSemaphore.release();
     }
 
     private void addToReceivedMessagesQueue(IotHubTransportMessage message)
     {
-        synchronized (this.receiveThreadLock)
-        {
-            this.receivedMessagesQueue.add(message);
+        this.receivedMessagesQueue.add(message);
 
-            // Wake up IotHubReceiveTask so it can handle receiving this message
-            this.receiveThreadLock.notifyAll();
-        }
+        // Wake up IotHubReceiveTask so it can handle receiving this message
+        this.receiveThreadSemaphore.release();
     }
 
     /**
-     * Sleep for a length of time without interruption
+     * Sleep for a length of time
      *
-     * @param sleepFor length of time to sleep for
-     * @param unit time unit associated with sleepFor
+     * @param sleepFor length of time to sleep for, in milliseconds
      */
-    @SuppressWarnings("SameParameterValue")
-    // The TimeUnit is currently always MilliSeconds, but this method can be used generically as well.
-    private static void sleepUninterruptibly(long sleepFor, TimeUnit unit)
```

Comment:
Not sure why we wanted the retry logic to be uninterruptable, but now we do want it to be interruptable if the client is closing, so we can get rid of this method and replace it with a simple Thread.sleep call

Reply:
Obviously, you've never cared for an infant. Everyone benefits from uninterrupted sleep!